engine/classes/ElggEntity.php

```diff
@@ -1686,7 +1686,14 @@ protected function update() {
 		unset($persisted_entity);
 
 		if ($allow_edit) {
-			$allow_edit = _elgg_services()->events->trigger('update', $this->type, $this);
+			$allow_edit = elgg_trigger_before_event('update', $this->type, $this);
```

Comment:
So much for less intrusive deprecation behaviour ;) I don't think any of the older events would need to use :before, so maybe there is a way to make this work without deprecation warning.